python/dgl/transforms/functional.py

```diff
@@ -3300,4 +3302,104 @@ def norm_by_dst(g, etype=None):
 
     return norm
 
+def rwpe(g, k):
+    r"""Random Walk Positional Encoding, as introduced in
+    `Graph Neural Networks with Learnable Structural and Positional Representations
+    <https://arxiv.org/abs/2110.07875>`__
+
+    This function computes the random walk positional encodings as landing probabilities
+    from 1-step to k-step, starting from each node to itself.
+
+    Parameters
+    ----------
+    g : DGLGraph
+        The homogeneous graph.
+    k : int
+        The number of random walk steps. The paper found the best value to be 16 and 20
+        for two experiments.
+
+    Returns
+    -------
+    Tensor
+        The random walk positional encodings.
+
+    Example
+    -------
+    >>> import dgl
+    >>> g = dgl.graph(([0,1,1], [1,1,0]))
+    >>> dgl.rwpe(g, 2)
+    tensor([[0.0000, 0.5000],
+            [0.5000, 0.7500]])
+    """
+    # get 1-step random walk probabilities as A * D^-1
+    A = g.adj(scipy_fmt='csr') # adjacency matrix
+    Dinv = sparse.diags(F.asnumpy(g.in_degrees()).clip(1) ** -1.0, dtype=float) # D^-1
+    RW = A * Dinv
+
+    # Iterate for k steps
+    PE = [F.tensor(RW.diagonal()).float()]
+    RW_power = RW
+    for _ in range(k-1):
+        RW_power = RW_power * RW
+        PE.append(F.tensor(RW_power.diagonal()).float())
+    PE = F.stack(PE,dim=-1)
+    
+    return PE
+
+def lappe(g, k):
+    r"""Laplacian Positional Encoding, as introduced in
+    `A Generalization of Transformer Networks to Graphs
+    <https://arxiv.org/abs/2012.09699>`__
```

Comment:
It seems that Benchmarking Graph Neural Networks (https://arxiv.org/abs/2003.00982) is an earlier work that introduces this idea by the same group of people. We should reference this paper instead. My bad, sorry.

Reply:
Yeah, make sure we are citing the right paper!

Reply:
fixed